test/core/end2end/fixtures/h2_full_no_retry.cc

```diff
@@ -0,0 +1,118 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <string.h>
+
+#include <grpc/grpc_security.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/sync.h>
+
+#include "src/core/ext/filters/client_channel/client_channel.h"
+#include "src/core/ext/filters/http/server/http_server_filter.h"
+#include "src/core/ext/transport/chttp2/transport/chttp2_transport.h"
+#include "src/core/lib/channel/connected_channel.h"
+#include "src/core/lib/gprpp/host_port.h"
+#include "src/core/lib/surface/channel.h"
+#include "src/core/lib/surface/server.h"
+#include "test/core/end2end/end2end_tests.h"
+#include "test/core/util/port.h"
+#include "test/core/util/test_config.h"
+
+struct fullstack_fixture_data {
+  std::string localaddr;
+};
+
+static grpc_end2end_test_fixture chttp2_create_fixture_fullstack(
+    const grpc_channel_args* /*client_args*/,
+    const grpc_channel_args* /*server_args*/) {
+  grpc_end2end_test_fixture f;
+  int port = grpc_pick_unused_port_or_die();
+  fullstack_fixture_data* ffd = new fullstack_fixture_data();
+  memset(&f, 0, sizeof(f));
+
+  ffd->localaddr = grpc_core::JoinHostPort("localhost", port);
+
+  f.fixture_data = ffd;
+  f.cq = grpc_completion_queue_create_for_next(nullptr);
+
+  return f;
+}
+
+void chttp2_init_client_fullstack(grpc_end2end_test_fixture* f,
+                                  const grpc_channel_args* client_args) {
+  fullstack_fixture_data* ffd =
+      static_cast<fullstack_fixture_data*>(f->fixture_data);
+  grpc_channel_credentials* creds = grpc_insecure_credentials_create();
+  grpc_arg no_retry = grpc_channel_arg_integer_create(
+      const_cast<char*>(GRPC_ARG_ENABLE_RETRIES), 0);
```

Comment:
Should we maybe add this test fixture as a separate PR?  It seems like a significant enough change on its own.